qualtran/bloqs/factoring/ecc/ec_add_r.py

```diff
@@ -67,6 +84,49 @@ def signature(self) -> 'Signature':
             [Register('ctrl', QBit()), Register('x', QUInt(self.n)), Register('y', QUInt(self.n))]
         )
 
+    def build_composite_bloq(
+        self, bb: 'BloqBuilder', ctrl: Soquet, x: Soquet, y: Soquet
+    ) -> Dict[str, 'SoquetT']:
+        if isinstance(self.n, sympy.Expr):
+            raise DecomposeTypeError("Cannot decompose symbolic `n`.")
+
+        a = bb.add(IntState(bitsize=self.n, val=0))
+        b = bb.add(IntState(bitsize=self.n, val=0))
+
+        ctrl_spec = CtrlSpec(qdtypes=QBit(), cvs=(1))
+        ctrl, a = bb.add(
+            XorK(QUInt(bitsize=self.n), self.R.x).controlled(ctrl_spec=ctrl_spec), ctrl=ctrl, x=a
+        )
+        ctrl, b = bb.add(
+            XorK(QUInt(bitsize=self.n), self.R.y).controlled(ctrl_spec=ctrl_spec), ctrl=ctrl, x=b
+        )
+
+        lam_num = (3 * self.R.x**2 + self.R.curve_a) % self.R.mod
+        lam_denom = (2 * self.R.y) % self.R.mod
+
+        lam = (lam_num * pow(lam_denom, -1, mod=self.R.mod)) % self.R.mod
+        lam_r = bb.add(IntState(bitsize=self.n, val=lam))
+
+        a, b, x, y, lam_r = bb.add(ECAdd(self.n, self.R.mod), a=a, b=b, x=x, y=y, lam=lam_r)
+
+        # Call graph doesn't like the controlled() operator on arbitrary clifford.
+        ctrl, a = bb.add(
+            XorK(QUInt(bitsize=self.n), self.R.x).controlled(ctrl_spec=ctrl_spec), ctrl=ctrl, x=a
+        )
+        ctrl, b = bb.add(
+            XorK(QUInt(bitsize=self.n), self.R.y).controlled(ctrl_spec=ctrl_spec), ctrl=ctrl, x=b
+        )
+
+        bb.add(Free(QUInt(self.n)), reg=a)
+        bb.add(Free(QUInt(self.n)), reg=b)
+        bb.add(Free(QUInt(self.n)), reg=lam_r)
+
+        return {'ctrl': ctrl, 'x': x, 'y': y}
+
+    def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
```

Comment:
although this is still supported for backwards compatibility, we've transitioned to returning a dictionary for this method. can you please update